+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Thu Dec 29 00:28:38 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL 'mailman postfix'
+ pkgs='mailman postfix'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs 'mailman postfix' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpulki0tqg
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpulki0tqg
+ old_LANG=
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ expect
spawn /usr/lib/mailman/bin/newlist mailman
couldn't execute "/usr/lib/mailman/bin/newlist": no such file or directory
    while executing
"spawn /usr/lib/mailman/bin/newlist mailman"
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Thu Dec 29 00:32:03 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Thu Dec 29 00:32:04 2022 - INFO  - Start testing...
+ test_execution mailman.service
+ service=mailman.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-12-29 00:32:04'
+ test_restart mailman.service
+ service=mailman.service
+ systemctl restart mailman.service
Failed to restart mailman.service: Unit mailman.service not found.
+ CHECK_RESULT 5 0 0 'mailman.service restart failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='mailman.service restart failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'mailman.service restart failed'
+ LOG_ERROR 'mailman.service restart failed'
+ message='mailman.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service restart failed'
Thu Dec 29 00:32:05 2022 - ERROR - mailman.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Thu Dec 29 00:32:06 2022 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status mailman.service
+ grep 'Active: active'
Unit mailman.service could not be found.
+ CHECK_RESULT 1 0 0 'mailman.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='mailman.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'mailman.service restart failed'
+ LOG_ERROR 'mailman.service restart failed'
+ message='mailman.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service restart failed'
Thu Dec 29 00:32:13 2022 - ERROR - mailman.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Thu Dec 29 00:32:14 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop mailman.service
Failed to stop mailman.service: Unit mailman.service not loaded.
+ CHECK_RESULT 5 0 0 'mailman.service stop failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='mailman.service stop failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'mailman.service stop failed'
+ LOG_ERROR 'mailman.service stop failed'
+ message='mailman.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service stop failed'
Thu Dec 29 00:32:15 2022 - ERROR - mailman.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 39'
+ message='../common/common_lib.sh line 39'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 39'
Thu Dec 29 00:32:16 2022 - ERROR - ../common/common_lib.sh line 39
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ grep 'Active: inactive'
+ systemctl status mailman.service
Unit mailman.service could not be found.
+ CHECK_RESULT 1 0 0 'mailman.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='mailman.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'mailman.service stop failed'
+ LOG_ERROR 'mailman.service stop failed'
+ message='mailman.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service stop failed'
Thu Dec 29 00:32:23 2022 - ERROR - mailman.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Thu Dec 29 00:32:23 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start mailman.service
Failed to start mailman.service: Unit mailman.service not found.
+ CHECK_RESULT 5 0 0 'mailman.service start failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='mailman.service start failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'mailman.service start failed'
+ LOG_ERROR 'mailman.service start failed'
+ message='mailman.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service start failed'
Thu Dec 29 00:32:24 2022 - ERROR - mailman.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Thu Dec 29 00:32:25 2022 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status mailman.service
+ grep 'Active: active'
Unit mailman.service could not be found.
+ CHECK_RESULT 1 0 0 'mailman.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='mailman.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'mailman.service start failed'
+ LOG_ERROR 'mailman.service start failed'
+ message='mailman.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service start failed'
Thu Dec 29 00:32:32 2022 - ERROR - mailman.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Thu Dec 29 00:32:33 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled mailman.service
+ service=mailman.service
++ systemctl is-enabled mailman.service
Failed to get unit file state for mailman.service: No such file or directory
+ state=
+ '[' '' == enabled ']'
+ '[' '' == disabled ']'
+ '[' '' == masked ']'
+ '[' '' == static ']'
+ LOG_INFO 'Unit is indirect, ignoring.'
+ message='Unit is indirect, ignoring.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Unit is indirect, ignoring.'
Thu Dec 29 00:32:34 2022 - INFO  - Unit is indirect, ignoring.
+ journalctl --since '2022-12-29 00:32:04' -u mailman.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
+ CHECK_RESULT 1 0 1 'There is an error message for the log of mailman.service'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of mailman.service'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ systemctl start mailman.service
Failed to start mailman.service: Unit mailman.service not found.
+ sed -i 's\ExecStart=/usr/lib/mailman/bin/mailmanctl\ExecStart=/usr/lib/mailman/bin/mailmanctl -q\g' /usr/lib/systemd/system/mailman.service
sed: can't read /usr/lib/systemd/system/mailman.service: No such file or directory
+ systemctl daemon-reload
+ systemctl reload mailman.service
Failed to reload mailman.service: Unit mailman.service not found.
+ CHECK_RESULT 5 0 0 'mailman.service reload failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='mailman.service reload failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'mailman.service reload failed'
+ LOG_ERROR 'mailman.service reload failed'
+ message='mailman.service reload failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service reload failed'
Thu Dec 29 00:32:39 2022 - ERROR - mailman.service reload failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_mailman.sh line 51'
+ message='oe_test_service_mailman.sh line 51'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_mailman.sh line 51'
Thu Dec 29 00:32:40 2022 - ERROR - oe_test_service_mailman.sh line 51
+ return 0
+ systemctl status mailman.service
+ grep 'Active: active'
Unit mailman.service could not be found.
+ CHECK_RESULT 1 0 0 'mailman.service reload causes the service status to change'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='mailman.service reload causes the service status to change'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'mailman.service reload causes the service status to change'
+ LOG_ERROR 'mailman.service reload causes the service status to change'
+ message='mailman.service reload causes the service status to change'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'mailman.service reload causes the service status to change'
Thu Dec 29 00:32:41 2022 - ERROR - mailman.service reload causes the service status to change
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_mailman.sh line 53'
+ message='oe_test_service_mailman.sh line 53'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_mailman.sh line 53'
Thu Dec 29 00:32:42 2022 - ERROR - oe_test_service_mailman.sh line 53
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Thu Dec 29 00:32:43 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Dec 29 00:32:43 2022 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Thu Dec 29 00:32:44 2022 - INFO  - start environment cleanup.
+ sed -i 's\ExecStart=/usr/lib/mailman/bin/mailmanctl -q\ExecStart=/usr/lib/mailman/bin/mailmanctl\g' /usr/lib/systemd/system/mailman.service
sed: can't read /usr/lib/systemd/system/mailman.service: No such file or directory
+ systemctl daemon-reload
+ systemctl reload mailman.service
Failed to reload mailman.service: Unit mailman.service not found.
+ systemctl stop mailman.service
Failed to stop mailman.service: Unit mailman.service not loaded.
+ /usr/lib/mailman/bin/rmlist mailman
oe_test_service_mailman.sh: line 63: /usr/lib/mailman/bin/rmlist: No such file or directory
+ export LANG=
+ LANG=
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Thu Dec 29 00:32:49 2022 - INFO  - Finish environment cleanup!
